ledger-core/virtual/integration/method_test.go

```diff
@@ -532,3 +532,151 @@ func TestVirtual_CallContractFromContract_Unordered(t *testing.T) {
+		return execution.Request.Callee.String()
+	})
+	server.ReplaceRunner(runnerMock)
+	server.Init(ctx)
+	server.IncrementPulseAndWaitIdle(ctx)
+
+	var (
+		flags = contract.MethodIsolation{Interference: contract.CallTolerable, State: contract.CallDirty}
+
+		objectAGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+
+		classB         = gen.UniqueReference()
+		objectB1Global = reference.NewSelf(server.RandomLocalWithPulse())
+		objectB2Global = reference.NewSelf(server.RandomLocalWithPulse())
+		objectB3Global = reference.NewSelf(server.RandomLocalWithPulse())
+	)
+
+	Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+	Method_PrepareObject(ctx, server, payload.Ready, objectB1Global)
+	Method_PrepareObject(ctx, server, payload.Ready, objectB2Global)
+	Method_PrepareObject(ctx, server, payload.Ready, objectB3Global)
+
+	builder := executionevent.NewRPCBuilder(gen.UniqueReference(), objectAGlobal)
+	objectAExecutionMock := runnerMock.AddExecutionMock(objectAGlobal.String())
+	objectAExecutionMock.AddStart(
+		func(ctx execution.Context) {
+			t.Log("ExecutionStart [A.Foo]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:     executionupdate.OutgoingCall,
+			Error:    nil,
+			Outgoing: builder.CallMethod(objectB1Global, classB, "Bar", []byte{}),
+		},
+	)
+
+	objectAExecutionMock.AddContinue(
+		func(result []byte) {
+			t.Log("ExecutionContinue [A.Foo]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:     executionupdate.OutgoingCall,
+			Error:    nil,
+			Outgoing: builder.CallMethod(objectB2Global, classB, "Bar", []byte{}),
+		},
+	)
+	objectAExecutionMock.AddContinue(
+		func(result []byte) {
+			t.Log("ExecutionContinue [A.Foo]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:     executionupdate.OutgoingCall,
+			Error:    nil,
+			Outgoing: builder.CallMethod(objectB3Global, classB, "Bar", []byte{}),
+		},
+	)
+	objectAExecutionMock.AddContinue(
+		func(result []byte) {
+			t.Log("ExecutionContinue [A.Foo]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:   executionupdate.Done,
+			Error:  nil,
+			Result: requestresult.New([]byte("finish A.Foo"), objectAGlobal),
+		},
+	)
+
+	runnerMock.AddExecutionMock(objectB1Global.String()).AddStart(
+		func(ctx execution.Context) {
+			t.Log("ExecutionStart [B1.Bar]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:   executionupdate.Done,
+			Result: requestresult.New([]byte("finish B1.Bar"), objectB1Global),
+		},
+	)
+
+	runnerMock.AddExecutionMock(objectB2Global.String()).AddStart(
+		func(ctx execution.Context) {
+			t.Log("ExecutionStart [B2.Bar]")
+		},
+		&executionupdate.ContractExecutionStateUpdate{
+			Type:   executionupdate.Done,
+			Result: requestresult.New([]byte("finish B2.Bar"), objectB2Global),
+		},
+	)
+
+	runnerMock.AddExecutionMock(objectB3Global.String()).AddStart(
+		func(ctx execution.Context) {
+			t.Log("ExecutionStart [B3.Bar]")
```

Comment:
nothing is checked here except fact of the call

Reply:
done